manim/mobject/geometry/labeled_shapes.py

```diff
@@ -0,0 +1,151 @@
+r"""Mobjects that inherit from lines and contain a label along the length."""
+
+from __future__ import annotations
+
+__all__ = ["LabeledLine", "LabeledArrow"]
+
+from manim.constants import *
+from manim.mobject.geometry.line import Arrow, Line
+from manim.mobject.geometry.shape_matchers import (
+    BackgroundRectangle,
+    SurroundingRectangle,
+)
+from manim.utils.color import *
+
+
+class LabeledLine(Line):
+    """Constructs a line containing a label box somewhere along its length.
+
+    Parameters
+    ----------
+    label : str | Tex | MathTex | Text
+        Label that will be displayed on the line.
+    label_position : float | optional
+        A ratio in the range [0-1] to indicate the position of the label with respect to the length of the line. Default value is 0.5.
+    font_size : int | float | optional
+        Control font size for the label. This parameter is only used when `label` is of type `str`.
+    label_color: numpy.ndarray | optional
+        The color of the label's text. This parameter is only used when `label` is of type `str`.
+    label_frame : Bool | optional
+        Add a `SurroundingRectangle` frame to the label box.
+    frame_fill_color : numpy.ndarray | optional
+        Background color to fill the label box. If no value is provided, the background color of the canvas will be used.
+    frame_fill_opacity : float | optional
+        Determine the opacity of the label box by passing a value in the range [0-1], where 0 indicates complete transparency and 1 means full opacity.
+
+    .. seealso::
+        :class:`LabeledArrow`
+
+    Examples
+    --------
+    .. manim:: LabeledLineExample
+        :save_last_frame:
+
+        class LabeledLineExample(Scene):
+            def construct(self):
+                line = LabeledLine(
+                    label          = '0.5',
+                    label_position = 0.8,
+                    font_size      = 20,
+                    label_color    = WHITE,
+                    label_frame    = True,
+
+                    start=LEFT+DOWN,
+                    end=RIGHT+UP)
+
+
+                line.set_length(line.get_length() * 2)
+                self.add(line)
+    """
+
+    def __init__(
+        self,
+        label,
+        label_position: float = 0.5,
+        font_size=15,
+        label_color=WHITE,
+        label_frame=True,
+        frame_fill_color=None,
+        frame_fill_opacity=1,
```

Comment:
I'm not exactly sure how we want to type hint colors, maybe @behackl / @ad-chaos have feedback. In 3b1b's version they use `ManimColor = Union[Color | str | None]`. But seeing as this isn't done anywhere else in the library, we can maybe pass on label_color and frame_fill_color for now.
```suggestion
        label: str | Tex | MathTex | Text,
        label_position: float = 0.5,
        font_size: float = 15,
        label_color: Color | str | None = WHITE,
        label_frame: bool = True,
        frame_fill_color: Color | str | None = None,
        frame_fill_opacity: float = 1,
```